release/pkg/generate_spec.go

```diff
@@ -492,6 +492,23 @@ func (r *ReleaseConfig) GetSourceImageURI(name, repoName string, tagOptions map[
 
 func (r *ReleaseConfig) GetSourceHelmURI(repoName string) (string, error) {
 	var sourceImageUri string
+	// If it's running a production release we only lookup from ECR public of the staging account.
+	if r.ReleaseEnvironment == "production" {
+		ecrpublicClient, err := NewECRPublicClient()
```

Comment:
We probably don't need to create new clients. The release config struct has source clients and release clients configured in it, for ecr, ecr public and s3

Reply:
This is using the SDK v2 so we might need to do so.